python/strings_udf/strings_udf/lowering.py

```diff
@@ -0,0 +1,637 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import operator
+
+from numba import cuda, types
+from numba.core import cgutils
+from numba.core.typing import signature as nb_signature
+from numba.cuda.cudadrv import nvvm
+from numba.cuda.cudaimpl import (
+    lower as cuda_lower,
+    registry as cuda_lowering_registry,
+)
+
+from strings_udf._lib.tables import get_character_flags_table_ptr
+from strings_udf._typing import size_type, string_view
+
+character_flags_table_ptr = get_character_flags_table_ptr()
+
+
+_string_view_len = cuda.declare_device(
+    "len", size_type(types.CPointer(string_view))
+)
+
+_string_view_contains = cuda.declare_device(
+    "contains",
+    types.boolean(types.CPointer(string_view), types.CPointer(string_view)),
+)
+
+
+_string_view_eq = cuda.declare_device(
+    "eq",
+    types.boolean(types.CPointer(string_view), types.CPointer(string_view)),
+)
+_string_view_ne = cuda.declare_device(
+    "ne",
+    types.boolean(types.CPointer(string_view), types.CPointer(string_view)),
+)
+
+_string_view_ge = cuda.declare_device(
+    "ge",
+    types.boolean(types.CPointer(string_view), types.CPointer(string_view)),
+)
+
+_string_view_le = cuda.declare_device(
+    "le",
+    types.boolean(types.CPointer(string_view), types.CPointer(string_view)),
+)
+_string_view_gt = cuda.declare_device(
+    "gt",
+    types.boolean(types.CPointer(string_view), types.CPointer(string_view)),
+)
+
+_string_view_lt = cuda.declare_device(
+    "lt",
+    types.boolean(types.CPointer(string_view), types.CPointer(string_view)),
+)
+
+_string_view_startswith = cuda.declare_device(
+    "startswith",
+    types.boolean(types.CPointer(string_view), types.CPointer(string_view)),
+)
+
+_string_view_endswith = cuda.declare_device(
+    "endswith",
+    types.boolean(types.CPointer(string_view), types.CPointer(string_view)),
+)
+
+_string_view_find = cuda.declare_device(
+    "find",
+    size_type(types.CPointer(string_view), types.CPointer(string_view)),
+)
+
+_string_view_rfind = cuda.declare_device(
+    "rfind",
+    size_type(types.CPointer(string_view), types.CPointer(string_view)),
+)
+
+_string_view_isdigit = cuda.declare_device(
+    "pyisdigit", types.boolean(types.CPointer(string_view), types.int64)
+)
+
+
```

Comment:
Spacing is inconsistent. 2 blank lines here vs. 1 elsewhere. Is this being run through pre-commit / formatters? `black` should fix this.